test/basic-implementation/BasicImplementation.ooc

```diff
@@ -0,0 +1,11 @@
+use flow
+use geometry
+use media
+
+main: func (arguments: String[]) {
+	input := Decoder new(arguments[2], IntVector2D parse(arguments[1]))
+	last := input as Filter
+	last connect(Displayer new())
+	input play()
+	exit(1)
```

Comment:
This `exit(1)` call never happens. It seems the only way to exit the application is by killing it, which results in an ugly stack trace. If the input stream is allowed to play to the end, the application will freeze on the final frame, and again there's no way to exit except by killing the application. This is problematic if you want to automate things or if you want to run valgrind to check for memory leaks.

Also, if `avconv` doesn't exist, the program will simply freeze. No error messages, nothing.

Do you want me to create issues for the above problems?

Reply:
It can wait as it works well enough for now. Long term we should probably test the video parameters first before starting the process. We can then also learn about the expected length and to dead counting. We can of course also figure out how to got too know that the process we started actually did shut down and stop polling the pipe.